backend/tracim_backend/views/core_api/schemas.py

```diff
@@ -1031,6 +1031,7 @@ class Meta:
 
 class WorkspaceMinimalSchema(marshmallow.Schema):
     workspace_id = marshmallow.fields.Int(example=4, validate=strictly_positive_int_validator)
+    description = StrippedString(example="All intranet data.")
```

Comment:
what the reason behind this change ? WorkspaceMinimalSchema is not really minimal anymore.

Reply:
I checked its usage, it is used only in `users/…/workspaces` and I think both APIs should return the same content. Also we already have 3 different public representations of a workspace (minimal,digest,full) adding one would make the usage of the APIs even more difficult (many `workspaces` endpoint but not two return the same structure).

Reply:
I also saw that the `/api/workspaces` endoint doesn't use the digest representation which surprised me a bit.

Reply:
https://github.com/tracim/tracim/issues/3601#issuecomment-694872894
Current usage of schema:
```
WorkspaceMinimalSchema:
- GET /workspaces/{workspace_id}/disk_space as subitem
WorkspaceDigestSchema :
- GET /users/{user_id:\d+}/workspaces
- GET /workspaces/{workspace_id}/members as subitem
- GET /workspaces/{workspace_id}/members/{user_id} as subitem
- PUT /workspaces/{workspace_id}/members/{user_id} as subitem
WorkspaceSchema:
- GET /workspaces
- POST /workspaces
- PUT /workspaces/{workspace_id} 
- GET /workspaces/{workspace_id}
- TLM
```

I don't see any advantage to get description in
```
- GET /workspaces/{workspace_id}/disk_space as subitem
- GET /workspaces/{workspace_id}/members as subitem
- GET /workspaces/{workspace_id}/members/{user_id} as subitem
- PUT /workspaces/{workspace_id}/members/{user_id} as subitem
```

Having description in
```
- GET /users/{user_id:\d+}/workspaces
```
may be not a big deal, has user has already access, and make things more consistent with `GET /workspaces`

So, i suggest:
```
WorkspaceMinimalSchema: same as actual
- GET /workspaces/{workspace_id}/disk_space as subitem
WorkspaceDigestSchema : all but sidebars entries
- GET /workspaces/{workspace_id}/members as subitem
- GET /workspaces/{workspace_id}/members/{user_id} as subitem
- PUT /workspaces/{workspace_id}/members/{user_id} as subitem
WorkspaceSchema:
- GET /users/{user_id:\d+}/workspaces
- GET /api/users/<user_id>/accessible_workspaces
- GET /workspaces
- POST /workspaces
- PUT /workspaces/{workspace_id} 
- GET /workspaces/{workspace_id}
- TLM
```

Reply:
@grignards can we clarify this point, you instead move description to DigestSchema, which mean description now added to :
```
- GET /workspaces/{workspace_id}/members as subitem
- GET /workspaces/{workspace_id}/members/{user_id} as subitem
- PUT /workspaces/{workspace_id}/members/{user_id} as subitem
```
which is not needed.

Reply:
I can return `WorkspaceSchema` in `users/<user_id>/(workspaces,accessible_workspaces) but in your proposal you remove `sidebar_entries` from `Digest` schema, are we sure that the frontend doesn't use this attribute from the routes using the `DigestSchema`? We should check what they are using from this route, if the `MinimalSchema` is enough for them we could remove/rename `Digest` to only keep two schemas for workspaces.

Reply:
Yes we are sure because members schema does not contain sidebar_entries:
```
class WorkspaceMemberSchema(WorkspaceMemberDigestSchema):
    user_id = marshmallow.fields.Int(example=3, validate=strictly_positive_int_validator)
    workspace_id = marshmallow.fields.Int(example=4, validate=strictly_positive_int_validator)
    is_active = marshmallow.fields.Bool()
    user = marshmallow.fields.Nested(UserDigestSchema())
    workspace = marshmallow.fields.Nested(WorkspaceDigestSchema(exclude=("sidebar_entries",)))

    class Meta:
        description = "Workspace Member information"
```

Also, my suggestion say Digest should contain all but sidebar entries, but in fact we can just remove sidebar entries from current digest and keep it like this.

But you're right, it may probably more sense to check what is useful for workspace subitem in workspace members endpoint in frontend.

So i suggest check usage of workspace subitem in sidebar entry and merge digest/minimal schema if possible (and do the other change i suggest to keep consistency).